GTScript-Parallel-model.md

```diff
@@ -1,13 +1,25 @@
-The iteration domain is a 3d domain: `I` and `J` axes live on the horizontal spatial plane, and axis `K` represents the vertical spatial dimension.
+The iteration domain is a 3d domain: `I` and `J` axes live on the horizontal spatial plane, and axis
+`K` represents the vertical spatial dimension.
 
-A `gtscript.stencil` is composed of one or more `computation`. Each `computation` defines an interation policy (`FORWARD`, `BACKWARD`, `PARALLEL`) and is itself composed of of one or more non-overlapping vertical `interval` specifications, each one of them representing a vertical loop over with the iteration policy of the coputation. Each interval contains one or more statements.
+A `gtscript.stencil` is composed of one or more `computation`. Each `computation` defines an
+interation policy (`FORWARD`, `BACKWARD`, `PARALLEL`) and is itself composed of of one or more
+non-overlapping vertical `interval` specifications, each one of them representing a vertical loop
+over the indicated indices of the `K`-axis, with the iteration policy of the coputation. Each
+interval contains one or more statements.
 
 The effect of the program is as if statements are executed as follows:
 
 - *computations* are executed sequentially in the order they appear in the code,
-- vertical *intervals* are executed sequentially in the order defined by the *iteration policy* of the *computation*
-- every vertical *interval* is executed as a sequential for-loop over the `K`-range following the order defined by the iteration policy,
-- every *statement* inside the *interval* is executed as a parallel for-loop over the horizontal dimension(s) with no guarantee on the order.
+- vertical *intervals* are executed sequentially in the order defined by the *iteration policy* of
+  the *computation*
+- every vertical *interval* is executed as a sequential for-loop over the `K`-range following the
+  order defined by the iteration policy,
+- every *statement* inside the *interval* is executed as a parallel for-loop over the horizontal
+  dimension(s) with no guarantee on the order.
+- the condition of if statements are evaluated where they are stated and saved to an internal
+  temporary field. Each statement of the if scope then only applied if this temporary field is
+  `True` at the respective grid point. Similarly, every statement in the else branch is checked,
+  but the condition is negated. 
```

Comment:
```suggestion
- every *statement* inside the *interval* (including statements inside conditional branches) is executed as a parallel for-loop over the horizontal
  dimension(s) with no guarantee on the order.

#### Implementation note:
A direction translation of conditionals requires the following pattern: the condition of if statements are evaluated where they are stated and saved to an internal
  temporary field. Each statement of the if scope is then only applied if this temporary field is
  `True` at the respective grid point. Similarly, every statement in the else branch is checked with the negated condition. 
```

Reply:
I wouldn't complicate the parallel model with conditionals if we go with this approach. They are not a special case for the parallel model, only for the implementation.